src/udiff/_vjp_diffs.py

```diff
@@ -553,12 +553,12 @@ def matmul_adjoint_1(A, G, A_ndim, B_meta):
 
 def matmul_vjp_0(ans, A, B):
     A_meta = metadata(A)
-    B_ndim = np.ndim(B)
+    B_ndim = B.ndim
```

Comment:
Why are these changes necessary?

Reply:
If we use `np.ndim`, it will return a `DiffArray`, which will cause an error. There may be a bug. I'm trying to figure out how to fix it.

Reply:
The error is caused by `return NotImplemented`. When we call `np.ndim(B)`, `B` is a raw `np.array`, and `coerce` is `False`, so the function `__ua_convert__` return `NotImplemented`.

Reply:
Can you provide a script?

Reply:
I have fixed it. The error is caused by the code in `register_vjp`:
```python
self._vjp = vjpmaker(tuple(parent_argnums), self.value, tuple(vjp_args), kwargs)
```

Reply:
This is still using `.ndim` instead of `np.ndim`. Did you forget to push?